dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```diff
@@ -0,0 +1,673 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.config.admin.internal;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.ibm.ws.config.admin.ConfigID;
+import com.ibm.ws.config.admin.ConfigurationDictionary;
+import com.ibm.ws.config.admin.ExtendedConfiguration;
+import com.ibm.wsspi.kernel.service.utils.OnErrorUtil.OnError;
+import com.ibm.wsspi.kernel.service.utils.SerializableProtectedString;
+
+/**
+ * A helper class for storing and retrieving the persisted config information.
+ *
+ * The format is:
+ *
+ * byte - version, set to 0
```

Comment:
The current code is out of sync with this description. So far all I can say for sure is that after the version an int number of configs is written.

Reply:
Updated the comments with the format.  This made me realize that I did not store the version for the Map used for the variable registry cache.  This is now fixed.